pandas/core/indexes/base.py

```diff
@@ -4197,17 +4206,19 @@ def _get_leaf_sorter(labels):
                 mask = new_lev_codes != -1
                 if not mask.all():
                     new_codes = [lab[mask] for lab in new_codes]
-                    left_indexer = left_indexer[mask]
+                    # error: Value of type "Optional[ndarray]" is not indexable
+                    left_indexer = left_indexer[mask]  # type: ignore[index]
 
             else:  # tie out the order with other
                 if level == 0:  # outer most level, take the fast route
                     ngroups = 1 + new_lev_codes.max()
                     left_indexer, counts = libalgos.groupsort_indexer(
-                        new_lev_codes, ngroups
+                        ensure_int64(new_lev_codes), ngroups
                     )
 
                     # missing values are placed first; drop them!
-                    left_indexer = left_indexer[counts[0] :]
+                    # error: Value of type "Optional[ndarray]" is not indexable
+                    left_indexer = left_indexer[counts[0] :]  # type: ignore[index]
```

Comment:
you could maybe leave this one as a "fix later" ignore instead of a cast as we can fix ourselves by typing `libalgos.groupsort_indexer` (i.e. adding algos.pyi)